devito/ir/equations/algorithms.py

```diff
@@ -95,18 +95,33 @@ def lower_exprs(expressions, **kwargs):
                   for f in retrieve_functions(expr)}
 
         # Handle Indexeds (from index notation)
-        for i in retrieve_indexed(expr):
+        for i in retrieve_indexed(expr, deep=True):
             f = i.function
 
             # Introduce shifting to align with the computational domain
             indices = [(a + o) for a, o in zip(i.indices, f._size_nodomain.left)]
 
+            tmp_mapper = {}
+            # Indexify indices of nested functions
+            for idx, index in enumerate(indices):
+                if index.is_Function and not index.is_Dimension and not index.is_Indexed:
```

Comment:
no, this is quite hacky

Reply:
Yes, we agreed. However, it seems to be working for deep nests. Thoughts? 
If it works why not? 
Hack removed by using `retrieve_functions`